functions.php

```diff
@@ -115,9 +115,15 @@ function as_unschedule_action( $hook, $args = array(), $group = '' ) {
 		$params['args'] = $args;
 	}
 
-	$action_id = ActionScheduler::store()->query_action( $params );
-	if ( $action_id ) {
-		ActionScheduler::store()->cancel_action( $action_id );
+	try {
+		$action_id = ActionScheduler::store()->query_action( $params );
+		if ( $action_id ) {
+			ActionScheduler::store()->cancel_action( $action_id );
+		}
+	} catch ( Exception $exception ) {
+		$message = sprintf( __( 'Caught exception while cancelling action: %s', 'action-scheduler' ), esc_attr( $hook ) );
+		error_log( $message, E_WARNING );
+		$action_id = null;
 	}
```

Comment:
Thinking we should be consistent in where we log things, and also because I think it is safe to move the `query_action()` call out of the `try {}` structure (afaik, it does not itself throw any exceptions), what do you think of the following tweaks?

```suggestion
	$action_id = ActionScheduler::store()->query_action( $params );

	if ( $action_id ) {
		try {
			ActionScheduler::store()->cancel_action( $action_id );
		} catch ( Exception $exception ) {
			ActionScheduler::logger()->log(
				$action_id,
				sprintf(
					/* translators: %s is the name of the hook to be cancelled. */
					__( 'Caught exception while cancelling action: %s', 'action-scheduler' ),
					esc_attr( $hook )
				)
			);

			$action_id = null;
		}
	}
```

Almost the same, but uses Action Scheduler's own logger (and some minor stuff, like a translator note).